imaids/shimming.py

```diff
@@ -137,22 +138,112 @@ def __init__(
         if zmax_pe is None:
             zmax_pe = zmax
 
-        self.cassettes = cassettes
-        self.block_type = block_type
-        self.segments_type = segments_type
-        self.zmin = zmin
-        self.zmax = zmax
-        self.znpts = int(znpts)
-        self.energy = energy
-        self.rkstep = rkstep
-        self.xpos = xpos
-        self.ypos = ypos
-        self.zmin_pe = zmin_pe
-        self.zmax_pe = zmax_pe
-        self.include_pe = include_pe
-        self.field_comp = field_comp
-        self.solved_shim = solved_shim
-        self.solved_matrix = solved_matrix
+        self._zmin = zmin
+        self._zmax = zmax
+        self._znpts = int(znpts)
+        self._cassettes = cassettes
+        self._block_type = block_type
+        self._segments_type = segments_type
+        self._energy = energy
+        self._rkstep = rkstep
+        self._xpos = xpos
+        self._ypos = ypos
+        self._zmin_pe = zmin_pe
+        self._zmax_pe = zmax_pe
+        self._include_pe = include_pe
+        self._field_comp = field_comp
+        self._solved_shim = solved_shim
+        self._solved_matrix = solved_matrix
+
+    @property
+    def zmin(self):
+        """Initial longitudinal position [mm]."""
+        return self._zmin
+    
+    @property
+    def zmax(self):
+        """Final longitudinal position [mm]."""
+        return self._zmax
+    
+    @property
+    def znpts(self):
+        """Number of sampling points for field integrals."""
+        return self._znpts
+    
+    @property
+    def cassettes(self):
+        """List of strings specifying which cassettes will be used in shimming.
+        """
+        return _deepcopy(self._cassettes)
```

Comment:
The cassette property at the undulator objects returns the cassettes themselves in a dictionary like {'cs':<the actual cassette object>, 'ci':<the actual cassette object>}. So returning a copy makes sense in some cases.
Here, the cassettes are only a list of strings, representing which cassettes will be used for shimming. So I believe there should not be a deepcopy here.